From 2257d1eadd02d89d225fce21013a1219d221dc7d Mon Sep 17 00:00:00 2001
From: dirk <dirk@aa41f4f7-0bf4-0310-aa73-e5a19afd5a74>
Date: Tue, 16 Dec 2014 12:17:20 +0000
Subject: [PATCH] Thread limit should be at least 1.

Limit thread when thread limit is 0. It is a logic error that could lead to ressource exhaustion.

git-svn-id: https://subversion.imagemagick.org/subversion/ImageMagick/branches/ImageMagick-6@17282 aa41f4f7-0bf4-0310-aa73-e5a19afd5a74
origin:  http://trac.imagemagick.org/changeset/17282
---
 magick/resource.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/magick/resource.c b/magick/resource.c
index 2f35520..2d79e6e 100644
--- a/magick/resource.c
+++ b/magick/resource.c
@@ -1193,6 +1193,8 @@ MagickExport MagickBooleanType SetMagickResourceLimit(const ResourceType type,
           100.0));
       if (resource_info.throttle_limit > GetOpenMPMaximumThreads())
         resource_info.throttle_limit=GetOpenMPMaximumThreads();
+      else if (resource_info.thread_limit == 0)
+        resource_info.thread_limit=1;
       break;
     }
     case TimeResource:
-- 
2.1.4

